### PR TITLE
using var is not recommended in ECMAscript

### PR DESCRIPTION
**Pull request by StEvUgnIn**
Using no-var-keyword rule and also marked a couple flaws with FIXME.

### CODE REVIEW

**Review by trusktr — APPROVED** [review]
🎉

**Comment by trusktr** [comment]
Needs an update

File: src/as/math/interpolants/QuaternionInterpolant.ts [hunk 1]
```diff
@@ -14,13 +14,14 @@ export class QuaternionLinearInterpolant extends Interpolant {
...
-			values = this.samplesValues,
+		let result = this.resultBuffer
+		const values = this.samplesValues,
 			stride = this.valueSize,
 			offset = i1 * stride,
 			alpha = (t - t0) / (t1 - t0)
 
-		for (var end = offset + stride; offset !== end; offset += 4) {
+		// FIXME
+		for (let end = offset + stride; offset !== end; offset += 4) {
```
Comment (trusktr) [inline src/as/math/interpolants/QuaternionInterpolant.ts:24]: The fix is small, I don't mind if you just stick it in here.

Comment (StEvUgnIn) [reply src/as/math/interpolants/QuaternionInterpolant.ts:24]: See my comment about it here https://github.com/mrdoob/three.js/commit/4126288ad3f53fcd9b6ebe4817fb3b674fd543c5

**Pull request merged**